### DISCUSSION TITLE
Dependant Filters

### DISCUSSION THREAD

**Discussion by futurewebsites** [area17/twill#1177] [General]
We currently have a categories filter which allows you to select a parent category this spawns another select box which contains all child categories and then on child selection another select box is spawned containing all grandchildren this continues to 4 or 5 levels if they exist products can filtered by parent or any sub level of the category tree.

Has anyone implemented anything like this or have any ideas if this behaviour could be replicated?

**Comment by pboivin** [comment]
Hi @futurewebsites, at a glance this definitely looks like custom Vue component territory. 

I'm thinking a good way to explore this would be to create a `SelectSequence` component, which would start with a `SingleSelect`, then create the next `SingleSelect` depending on the value of the first one, and so on.

Custom component workflows are not super well documented at this point but maybe this tutorial can help get you started:
https://spectrum.chat/twill/tips-and-tricks/adding-a-custom-block-to-twill-admin-view-with-vuejs~028d79b1-b3cd-4fb7-a89c-ce64af7be4af

The tutorial is focused on custom blocks, which is probably not what you want, but the workflow is identical for custom components.

**Reply by futurewebsites** [reply]
Thanks @pboivin
